R/USA.R

```diff
@@ -0,0 +1,119 @@
+    #' @field supported_levels A list of supported levels.
+    supported_levels = list("1", "2"),
+    #' @field supported_region_names A list of region names in order of level.
+    supported_region_names = list(
+      "1" = "state",
+      "2" = "county"
+    ),
+    #' @field supported_region_codes A list of region codes in order of level.
+    supported_region_codes = list(
+      "1" = "iso_3166_2",
+      "2" = "fips"
+    ),
+    #' @field common_data_urls List of named links to raw data. Since there are
+    #'  no common data sources for level 1 and 2, this list is empty.
+    common_data_urls = list(),
+    #' @field level_data_urls List of named links to raw data that are level
+    #' specific.
+    level_data_urls = list(
+      "1" = list("state" = "https://raw.githubusercontent.com/nytimes/covid-19-data/master/us-states.csv"), # nolint
+      "2" = list("county" = "https://raw.githubusercontent.com/nytimes/covid-19-data/master/us-counties.csv") # nolint
+    ),
+    #' @field source_data_cols existing columns within the raw data
+    source_data_cols = c("cases_total", "deaths_total"),
+
+    #' @description Set up a table of region codes for clean data
+    #' @importFrom tibble tribble
+    set_region_codes = function() {
+      self$codes_lookup$`1` <- usa_codes <- tibble(
+        level_1_region_code = c(
+          "US-AL", "US-AK", "US-AZ", "US-AR", "US-CA", "US-CO", "US-CT",
+          "US-DE", "US-FL", "US-GA", "US-HI", "US-ID", "US-IL", "US-IN",
+          "US-IA", "US-KS", "US-KY", "US-LA", "US-ME", "US-MD", "US-MA",
+          "US-MI", "US-MN", "US-MS", "US-MO", "US-MT", "US-NE", "US-NV",
+          "US-NH", "US-NJ", "US-NM", "US-NY", "US-NC", "US-ND", "US-OH",
+          "US-OK", "US-OR", "US-PA", "US-RI", "US-SC", "US-SD", "US-TN",
+          "US-TX", "US-UT", "US-VE", "US-VA", "US-WA", "US-WV", "US-WI",
+          "US-WY", "US-DC", "US-AS", "US-GU", "US-MP", "US-PR", "US-UM",
+          "US-VI"
+        ),
+        region = c(
+          "Alabama", "Alaska", "Arizona", "Arkansas", "California", "Colorado",
+          "Connecticut", "Delaware", "Florida", "Georgia", "Hawaii", "Idaho",
+          "Illinois", "Indiana", "Iowa", "Kansas", "Kentucky", "Louisiana",
+          "Maine", "Maryland", "Massachusetts", "Michigan", "Minnesota",
+          "Mississippi", "Missouri", "Montana", "Nebraska", "Nevada",
+          "New Hampshire", "New Jersey", "New Mexico", "New York",
+          "North Carolina", "North Dakota", "Ohio", "Oklahoma", "Oregon",
+          "Pennsylvania", "Rhode Island", "South Carolina", "South Dakota",
+          "Tennessee", "Texas", "Utah", "Vermont", "Virginia", "Washington",
+          "West Virginia", "Wisconsin", "Wyoming", "District of Columbia",
+          "American Samoa", "Guam", "Northern Mariana Islands", "Puerto Rico",
+          "Minor Outlying Islands", "Virgin Islands"
+        )
+      )
+      self$codes_lookup$`2` <- self$codes_lookup$`1`
```

Comment:
still confused by this.

Reply:
Yeah I think we can just drop the second codes lookup and just have the one we call